File: C++/Program/Kendaraan.cpp
/*

Saya Nabilla Assyfa Ramadhani [2205297] 
mengerjakan Latihan
dalam mata Desain dan Pemograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan 
seperti yang telah dispesifikasikan. 
Aamiin

*/

// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once

// Memasukan library yang digunakan
#include <iostream>
#include <string>


// Deklarasi namespace
using namespace std;

class Kendaraan {
    private :
        string nomor_plat;
        string merk;
        int tahun_produksi;
        string warna;

    public :

    //constructor
    Kendaraan (){

    }

    Kendaraan (string nomor_plat, string merk, int tahun, string warna){
        this->nomor_plat = nomor_plat;
        this->merk = merk;
        this->tahun_produksi = tahun;
        this->warna = warna;
    }

    // Metode get dan set untuk setiap atribut kendaraan

    void setPlat (string nomor){
        this->nomor_plat = nomor;
    }

    string getPlat (){
        return this->nomor_plat;
    }

    void setMerk (string merk){
        this->merk = merk;
    }

    string getMerk (){
        return this->merk;
    }

    void setTahun (int tahun){
        this->tahun_produksi = tahun;
    }

    int getTahun (){
        return this->tahun_produksi;
    }

    void setWarna (string warna){
        this->warna = warna;
    }

    string getWarna (){
        return this->warna;
    }


    //Destreuctor
    ~Kendaraan (){

    }


};

File: Python/Program/Kendaraan.py


# Saya Nabilla Assyfa Ramadhani [2205297] 
# mengerjakan Latihan
# dalam mata Desain dan Pemograman Berorientasi Objek 
# untuk keberkahanNya maka saya tidak melakukan kecurangan 
# seperti yang telah dispesifikasikan. 
# Aamiin


class Kendaraan :
    __nomor_plat =""   # variabel untuk menampung nomor plat kendaraan
    __merk =""          # variabel untuk menampung merk kendaraan
    __tahun_produksi= 0 # variabel untuk menampung tahun produksi suatu kendaraan
    __warna=""          # variabel untuk menampung nama kendaraan
        
        
    #constructor
    def __init__(self, nomor_plat,  merk, tahun,  warna):
       self.__nomor_plat = nomor_plat
       self.__merk = merk
       self.__tahun_produksi = tahun
       self.__warna = warna
    

    # Metode get dan set untuk setiap atribut kendaraan
    
    # metode set plat
    def setPlat (self, nomor):
       self.__nomor_plat = nomor
    # metode get plat
    def getPlat (self):
        return self.__nomor_plat
    

    def setMerk (self, merk):
       self.__merk = merk
       
    def getMerk (self):
        return self.__merk
    

    def setTahun (self, tahun):
       self.__tahun_produksi = tahun
       
    def getTahun (self):
        return self.__tahun_produksi
    

    def setWarna (self, warna):
       self.__warna = warna
   
    def getWarna (self):
        return self.__warna

    


